src/poetry/publishing/publisher.py

```diff
@@ -69,8 +72,8 @@ def publish(
                     logger.debug(
                         f"Found authentication information for {repository_name}."
                     )
-                    username = auth.username
-                    password = auth.password
+                    username = username or auth.username
+                    password = password or auth.password
```

Comment:
Unsure if we really have this scenario.

Reply:
This looks very niche to me as well. Do you think we should not consider this case?